src/main/java/it/gov/pagopa/payhub/activities/service/treasury/TreasuryOpiParserService.java

```diff
@@ -36,13 +57,50 @@ public List<String> parseData(Path ingestionFlowFilePath) {
             try {
                 flussoGiornaleDiCassa14 = treasuryUnmarshallerService.unmarshalOpi14(ingestionFlowFile);
                 log.debug("file flussoGiornaleDiCassa with Id {} parsed successfully ", flussoGiornaleDiCassa14.getId());
+                version = TreasuryVersionEnum.V_14;
             } catch (Exception exception) {
                 log.info("file flussoGiornaleDiCassa parsing error with opi 1.4 format {} ", exception.getMessage());
                 throw new TreasuryOpiInvalidFileException("Cannot parse treasury Opi file " + ingestionFlowFile);
             }
         }
-        //TODO in task 1658 it will be implemented the element to be returned
-        return List.of();
+
+
+        if (version == TreasuryVersionEnum.V_14) {
+            TreasuryValidatorService < FlussoGiornaleDiCassa> validatorService =
```

Comment:
You could use dependency injection to ask tò Spring for a Bean of type
private TreasuryValidatorService<...> 
Instead of having a class to hold instances